aicsimageio/readers/array_like_reader.py

```diff
@@ -0,0 +1,370 @@
+        A single, numpy ndarray, dask Array, or xarray DataArray, or list of many.
+        If provided a list, each item in the list will be exposed through the scene API.
+        If provided an xarray DataArray alone or as an element of the list, the
+        known_dims and known_channel_names, kwargs are ignored if there is dimension
+        (or channel coordinate) information attached the xarray object. If the provided
+        xarray object is missing these pieces, the AICSImageIO defaults will be added.
+
+    known_dims: Optional[Union[List[str], str]]
+        A string of known dimensions to be applied to all array(s) or a
+        list of string dimension names to be mapped onto the list of arrays
+        provided to image.
+        Default: None (guess dimensions for single array or multiple arrays)
+
+    known_channel_names: Optional[Union[List[str], List[List[str]]]]
+        A list of string channel names to be applied to all array(s) or a
+        list of lists of string channel names to be mapped onto the list of arrays
+        provided to image.
+        Default: None (create fake channel names for single or multiple arrays)
+
+    Raises
+    ------
+    exceptions.ConflictingArgumentsError: Raised when the number of scenes provided is
+    different from the number of items provided to the metadata parameters if as a list.
+
+    exceptions.ConflictingArgumentsError: Raised when known_channel_names is provided
+    but the channel dimension was either not guessed or not provided in known_dims.
+
+    ValueError: Provided known_dims string or known_channel_names are not the same
+    length as the number of dimensions or the size of the channel dimensions for the
+    array at the matching index.
+
+    Notes
+    -----
+    If you want to combine multiple numpy ndarrays or dask arrays with xarray DataArrays
+    and attach metadata to all non-xarray.DataArrays, you can do so, the metadata for
+    the xarray DataArray scenes will simply be ignored if it the non-xarray defaults.
+
+    Other way of saying this is:
+    If there are dimension names besides "dim_N" and there are coordinates
+    for the channel dimension, then the known_dims and known_channel_names will be
+    ignored. If the provided xarray object IS using the xarray defaults, we will
+    override their defaults.
+
+    In such cases, it is recommended that you provided metadata for those scenes as
+    None. I.E.
+
+    >>> some_xr = ...
+    ... some_np = ...
+    ... some_dask = ...
+    ... reader = ArrayLikeReader(
+    ...     image=[some_xr, some_np, some_dask],
+    ...     known_dims=[None, "CTYX", "ZTYX"],
+    ...     known_channel_names=[None, ["A", "B", C"], None],
+    ... )
+
+    Will create a three scene ArrayLikeReader with the raw xarray DataArray as the first
+    scene, a numpy array with CTYX as the dimensions and ["A", "B", "C"] as the
+    channel names, and a dask array with ZTYX as the dimensions and no channel names.
+    """
```

Comment:
Please give feedback on this docstring. I don't know how to explain all the functionality packed into this one reader so I think this is a bit long but feedback appreciated.